odoo/addons/base/models/ir_mail_server.py

```diff
@@ -702,7 +718,7 @@ def _find_mail_server(self, email_from, mail_servers=None):
         """
         email_from_normalized = email_normalize(email_from)
         email_from_domain = email_domain_extract(email_from_normalized)
-        notifications_email = email_normalize(self._get_default_from_address())
+        notifications_email = force_notifications_email or email_normalize(self._get_default_from_address())
```

Comment:
I think it will be better if it take the default email from the context (and so if we remove the `force_notifications_email` argument)

because currently, the `connect` method work properly only if we force the mail server (if we don't force it, it won't take the notification email from the context)

what do you think ?

(if you are ok with that, you can remove the `mail_server_id=mail_server.id` from the test above)